### ISSUE TITLE
Unescaped content sent on votes, leading to invalid forms

### ISSUE THREAD

**Issue by mtancoigne** [nextcloud/polls#341]
### Steps to reproduce
1. Create a new text based poll
2. Add an entry with double quotes (i.e: `Oui et j'ai un "petit" niveau (appels api, état global avec VueX, évènements)`
3. Add some entries and save
4. Visit the poll and change the state of the entry with double quotes.
5. Save your answers

### Expected behaviour
The entry with changed state should have its new state saved

### Actual behaviour
Status of the "double quoted string" did not change

### Server configuration

**Operating system**:
Ubuntu 16.04
**Web server:**
Apache 2
**Database:**
Mysql
**PHP version:**
7
**Nextcloud version:** (see Nextcloud admin page)
13.0.2
**Where did you install Nextcloud from:**
I don't remember
**Signing status:**

```
No errors have been found.
```

### Additional informations:
Going to "edit" the poll shows an empty list of options and the JS console states:

```
SyntaxError: JSON.parse: expected ',' or ']' after array element at line 1 column 9 of the JSON data
```

After _manually_ removing the quotes with a mysql client, everything went fine.

The issue comes from the fact that when a poll is submitted, the _raw_ values are sent to the server, unescaped, and when the server creates JSON for the edit form, the same thing happens...:

Data sent when voting for two options; `foo "foo"` and `bar bar`
```
changed | true
-- | --
dates | ["foo+","bar bar"]  // good strings would have been ["foo+\"foo\"", "bar bar"]
pollId | 4
receiveNotifications | false
types | [1,1]
userId | admin
```

### Possible things to do to correct this:
- Escape the strings before they are sent to server (and from server to edit form)
- better: send ids to server instead of strings...

**Comment by dartcafe** [comment]
Error in votes.js. Will inspect. Thanks for reporting

**Comment by joergmschulz** [comment]
Yes sir, something has still to be done in beta2 for 9.0.
Weird things might happen when you include html tags like:
<img width="333" alt="grafik" src="https://user-images.githubusercontent.com/6523646/42903636-ccb11a48-8ad2-11e8-9055-e9fe4f405eb3.png">


luckily, a  </php> tag has been thrown away just like HREF tags.

**Comment by joergmschulz** [comment]
browsers behave differently - while Safari and Firefox don't care about the "", Chrome does:
<img width="1047" alt="grafik" src="https://user-images.githubusercontent.com/6523646/42905461-6becf74e-8ad8-11e8-9342-7eaf5eb2c00c.png">

**Comment by dartcafe** [comment]
Fixed in 0.9-beta5
